src/d3a/d3a_core/rq_job_handler.py

```diff
@@ -25,6 +25,7 @@ def launch_simulation_from_rq_job(scenario, settings, events, aggregator_device_
                                   saved_state, job_id):
     logging.getLogger().setLevel(logging.ERROR)
     scenario = decompress_and_decode_queued_strings(scenario)
+    d3a.constants.SIMULATION_ID = job_id
```

Comment:
I think you can adapt it if you pass it as an extra argument in the activate() method of the area class. After all, this is the only place where the blockchain objects are constructed.